capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -42,21 +52,77 @@ export class WikiMediaComponent implements OnChanges, OnInit {
     let isOldRequest = false;
-    this.wikiService.search(queryString)
+    this.wikiService.search(queryString, 'en', 'History')
       .subscribe((result: WikiServiceResult) => {
         if (result.history && currentQuery === this.cityName) {
-            isOldRequest = true;
-            this.body = result.history;
-            this.history = result.history;
-            this.urls = result.furtherReading;
-            this.title = result.title;
-            this.loading = false;
-            this.cd.detectChanges();
+          this.body = result.history;
+          this.history = result.history;
+          this.urls = result.furtherReading;
+          this.title = result.title;
+          this.loading = false;
+          this.langlinks = result.langlinks;
+          const language = this.langlinks.get('ca');
+          this.prevQuery = language.get('searchQuery');
+          this.prevWordForHistory = 'History';
+          this.prevPreFix = 'en';
+          this.cd.detectChanges();
         }
         else if (!isOldRequest) {
           this.error = 'API did not return a valid response.';
           console.error('API did not return a valid response.');
         }
     });
   }
+
+  changeLanguage(queryString: string, languagePrefix: string, wordForHistory: string) {
+    this.wikiService.searchNewLang(queryString, languagePrefix, wordForHistory)
+      .subscribe((result: WikiServiceResult) => {
+        this.history = result.history;
+        console.log(this.history);
+        if (!(this.history)){
+          this.loading = true;
+          this.error = 'Unfortunately, the language you requested does not have an available tranlation '
+          + 'for this page. Please select a different language.';
+        }
+        else {
+          this.body = this.history;
+          this.loading = false;
+          this.title = result.title;
+          this.urls = result.furtherReading;
+          if(result.langlinks && result.langlinks.size > 0) {
+              this.langlinks = result.langlinks;
+          }
+        }
+        if (this.history.includes("CPU time usage:")){
+          this.loading = false;
+          this.error = 'Unfortunately, the language you requested has not been fully translated. '
+          + 'The data being shown is what the Wikipedia page returned.';
+        }
+    });
+  }
+
+  onChangeLanguage(prefix: string) {
+    if (prefix == this.prevPreFix) {
+      this.changeLanguage(this.prevQuery, this.prevPreFix, this.prevWordForHistory);
+    }
+    else if (prefix == 'en' && !(this.langlinks.get('en'))) {
+      this.prevPreFix = prefix;
+      const language = this.langlinks.get(prefix);
+      this.query = language.get('searchQuery');
+      this.prevQuery = this.query;
+      this.prevWordForHistory = 'History';
+      this.changeLanguage(this.query, 'en', 'History');
+    }
+    else {
+      const language = this.langlinks.get(prefix);
+      console.log(this.langlinks);
+      console.log(prefix);
```

Comment:
nit: remove console.logs

Reply:
Sorry about all of those, I totally forgot to remove them on my last commit.